Mathlib/Analysis/BoxIntegral/Partition/Basic.lean

```diff
@@ -742,9 +742,9 @@ theorem iUnion_subset (h : π.IsPartition) (π₁ : Prepartition I) : π₁.iUni
 #align box_integral.prepartition.is_partition.Union_subset BoxIntegral.Prepartition.IsPartition.iUnion_subset
 
 protected theorem existsUnique (h : π.IsPartition) (hx : x ∈ I) :
-    ∃! J, ∃! _ : J ∈ π, x ∈ J := by
+    ∃! J ∈ π, x ∈ J := by
   rcases h x hx with ⟨J, h, hx⟩
-  exact ExistsUnique.intro₂ J h hx fun J' h' hx' => π.eq_of_mem_of_mem h' h hx' hx
+  exact ExistsUnique.intro J ⟨h, hx⟩ fun J' ⟨h', hx'⟩ => π.eq_of_mem_of_mem h' h hx' hx
```

Comment:
Can you change the type of `ExistsUnique.intro₂` and `ExistsUnique.unique₂` to help this proof, and reflect the new standard form?

Reply:
I would rather not — I interpret these `ExistsUnique.intro₂` as being specifically about nested `∃!`'s, and I don't want to make the decision to change that API in this PR.

I'm also not sure that an adapted `ExistsUnique.intro₂` would help with this proof in any significant way. It'd just save five characters, right?